snmp/tests/test_unit.py

```diff
@@ -254,6 +254,11 @@ def test_cache_building(write_mock, read_mock):
 
     check = SnmpCheck('snmp', {}, [instance])
 
+    def fetch_error(cfg):
+        raise RuntimeError("fail")
+
+    check.fetch_sysobject_oid = fetch_error
```

Comment:
Can we clarify with a comment what problem this is solving, and why it's solving it? I'm not sure myself how this intertwines with cache writing. :confused: